modules/standard/DateTime.chpl

```diff
@@ -0,0 +1,1571 @@
+    return d + t;
+  }
+
+  pragma "no doc"
+  proc -(d: date, t: timedelta): date {
+    return date.fromordinal(d.toordinal() - t.days);
+  }
+
+  pragma "no doc"
+  proc -(d1: date, d2: date): timedelta {
+    return new timedelta(days=d1.toordinal() - d2.toordinal());
+  }
+
+  pragma "no doc"
+  proc <(d1: date, d2: date) {
+    return d1.toordinal() < d2.toordinal();
+  }
+
+  pragma "no doc"
+  proc <=(d1: date, d2: date) {
+    return d1.toordinal() <= d2.toordinal();
+  }
+
+  pragma "no doc"
+  proc >(d1: date, d2: date) {
+    return d1.toordinal() > d2.toordinal();
+  }
+
+  pragma "no doc"
+  proc >=(d1: date, d2: date) {
+    return d1.toordinal() >= d2.toordinal();
+  }
+
+
+  /* A record representing a time */
+  record time {
+    pragma "no doc"
+    var chpl_hour, chpl_minute, chpl_second, chpl_microsecond: int;
+    pragma "no doc"
+    var chpl_tzinfo: TZInfo;
+
+    /* The hour represented by this date value */
+    proc hour {
+      return chpl_hour;
+    }
+
+    /* The minute represented by this date value */
+    proc minute {
+      return chpl_minute;
+    }
+
+    /* The second represented by this date value */
+    proc second {
+      return chpl_second;
+    }
+
+    /* The microsecond represented by this date value */
+    proc microsecond {
+      return chpl_microsecond;
+    }
+
+    /* The timezone represented by this date value */
+    proc tzinfo {
+      return chpl_tzinfo;
+    }
+
+    /* The minimum representable time */
+    proc type min {
+      return new time();
+    }
+
+    /* The maximum representable time */
+    proc type max {
+      return new time(23, 59, 59, 999999);
+    }
+
+    /* The minimum non-zero difference between two times */
+    proc type resolution {
+      return new timedelta(microseconds=1);
+    }
+  }
+
+  /* constructors/factories for time values */
+
+  /* Construct a new time value from the given hour, minute, second,
+     microsecond, and timezone.  All arguments are optional
+   */
+  proc time.time(hour=0, minute=0, second=0, microsecond=0,
+                 tzinfo:TZInfo=nil) {
+    if hour < 0 || hour >= 24 then
+      halt("hour out of range");
+    if minute < 0 || minute >= 60 then
+      halt("minute out of range");
+    if second < 0 || second >= 60 then
+      halt("second out of range");
+    if microsecond < 0 || microsecond >= 1000000 then
+      halt("microsecond out of range");
+    this.chpl_hour = hour;
+    this.chpl_minute = minute;
+    this.chpl_second = second;
+    this.chpl_microsecond = microsecond;
+    this.chpl_tzinfo = tzinfo;
+  }
+
+  pragma "no doc"
+  proc time.~time() {
+    // delete tzinfo if needed
+  }
+
+  /* Methods on time values */
+
+  /* Replace the hour, minute, second, microsecond and tzinfo in a time
+     to create a new time. The tzinfo argument is required, the rest are
+     optional.
+   */
+  proc time.replace(hour=-1, minute=-1, second=-1, microsecond=-1,
+                    tzinfo:TZInfo) {
```

Comment:
Of course! I initially set it to `nil`, but then realized that there had to be a way to specify that you want to clear the `tzinfo`.  I didn't think of setting the default to `this.tzinfo`, so made it a required argument instead.